packages/storybook/src/nimble/text-field/text-field-matrix.stories.ts

```diff
@@ -92,6 +94,7 @@ export default metadata;
 const component = (
     [_readOnlyName, readonly]: ReadOnlyState,
     [_disabledName, disabled]: DisabledState,
+    [_appearanceReadOnlyName, appearanceReadonly]: AppearanceReadOnlyState,
```

Comment:
(open ended / brainstorming) Maybe we should merge the readonly, disabled, and appearanceReadonly states?
const editableStates =  [normal, readonly, disabled, appearanceReadonly]

names for the set of states could be: editableStates / interactionStates (we already use for hover and stuff so maybe not) / disabledReadonlyStates (very explicit)

Reply:
The downside I see with that approach is that we lose our test coverage for things like:
- `disabled` and `readOnly` - making sure `disabled` styling wins over `readOnly` styling
- not `disabled` and `appearanceReadOnly` - making sure to `appearanceReadOnly` causes no styling changes

Reply:
yea that's true, am alternate thought is add all those state permutations to collapse in a single dimension just so it's easier to split / add new dimensions in the future (particularly for text field since it has so many permutations / story splits). But I don't feel strongly there

Reply:
That turned out really nice for the storybook splits. We could maybe even remerge the button dimension but that's probably not even worth the effort to attempt. Looks good 👍